tests/setup.sh

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env bash
+
+set -eE
+trap 'die "failed setup"' ERR
+
+CURL="curl -fSsL"
+JQ="jq -r"
+
+echo_stderr() {
+  echo -e "${1}" >&2
+}
+
+die() {
+  local _return="${2:-1}"
+  test "${_PRINT_HELP:-no}" = yes && print_help >&2
+  echo_stderr "ERROR: $1"
+  exit "${_return}"
+}
+
+print_help() {
+  cat <<EOF
+Usage: $(basename "${0}") [OPTIONS]
+
+Options:
+  -h, --help                      Display this text.
+  --kubernetes-version <version>  Specify the version of kubernetes.
+  -i, --install-prerequisites     Install cri-dockerd and containernetworking plugins before creating minikube cluster.
+  -y, --assume-yes                Assume the answer 'Y' for all interactive questions.
+
+Examples:
+  $(basename "${0}") --kubernetes-version v1.25.10
+EOF
+}
+
+parse_args() {
+  while test $# -gt 0; do
+    arg="$1"
+    case "$arg" in
+      --kubernetes-version)
+        test $# -lt 2 && die "missing value for the optional argument '$arg'."
+        KUBERNETES_VERSION="${2}"
+        shift
+        ;;
+      --kubernetes-version=*)
+        KUBERNETES_VERSION="${arg#*=}"
+        ;;
+      -i | --install-prerequisites)
+        INSTALL_PREREQUISITES="true"
+        ;;
+      -y | --assume-yes)
+        ASSUME_YES="true"
+        ;;
+      -h | --help)
+        print_help
+        exit 0
+        ;;
+      -h*)
+        print_help
+        exit 0
+        ;;
+      *)
+        _PRINT_HELP=yes die "unexpected argument '$arg'" 1
+        ;;
+    esac
+    shift
+  done
+}
+
+must_succeed_command() {
+  local error="${2:-command \'${1}\' failed}"
+  ${1} || die "${error}"
+}
+
+must_exists_in_path() {
+  local error="${2:-command ${1} not present in PATH}"
+  must_succeed_command "command -v ${1}" "${error}"
+}
+
+get_latest_version_tag() {
+  local github_org=${1}
+  local github_repo=${2}
+
+  local tag_name=$(${CURL} -H "Accept: application/vnd.github+json" https://api.github.com/repos/${github_org}/${github_repo}/releases | ${JQ} '.[0].tag_name' | tr -d " \t\r\n")
+
+  echo "${tag_name}"
+}
+
+get_asset_url_from_latest_github_release() {
+  local github_org=${1}
+  local github_repo=${2}
+  local release_asset_regex=${3}
+
+  local url=$(${CURL} -H "Accept: application/vnd.github+json" https://api.github.com/repos/"${github_org}"/"${github_repo}"/releases | ${JQ} ".[0].assets[] | select(.name? | match(\"${release_asset_regex}\")).url" | tr -d " \t\r\n")
+
+  echo "${url}"
+}
+
+get_asset_binary() {
+  local github_asset_url=${1}
+  local workspace=${2}
+  local output_filepath=${3}
+
+  cd "${workspace}"
+  ${CURL} -H "Accept: application/octet-stream" "${github_asset_url}" -o "${output_filepath}"
+  cd -
+}
+
+install_prerequisites() {
+  # Check if LINUX.
+  local os_error="not a GNU/Linux environment"
+  os=$(must_succeed_command "uname -o" "${os_error}" | tr -d " \t\r\n")
+  if [ "${os}" != "GNU/Linux" ]; then
+    die "${os_error}"
```

Comment:
Doesn't run on the m1's ?

Reply:
No, it doesn't. For all practical uses with Mayastor, we'd have to stick to amd64 linux. But, it's a worthwhile enhancement for this script.  

I don't have a nix development environment on my m1, so I'd have to poke around to explore this :(

Reply:
The minikube part of the script will work on m1s, but not the prerequisites. There is one bug here, the kubectl cli tool's installation assumes it's an amd64. I'll fix it..

Reply:
Fixed it.